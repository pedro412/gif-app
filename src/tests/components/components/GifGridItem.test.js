import React from 'react';
import '@testing-library/jest-dom';
const { shallow } = require('enzyme');
const { default: GifGridItem } = require('../../../components/GifGridItem');

describe('testing <GifGridItem />', () => {
  const title = 'Hola';
  const url = 'https://algo.com';
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test('it should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('it should containt a paragraph with the title', () => {
    const p = wrapper.find('p');
    expect(p.text()).toBe(title);
  });

  test('it should contain an img tag with the src equal as the img url passed', () => {
    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(url);
  });

  test('it should contain the class grid-item', () => {
    const div = wrapper.find('div');
    expect(div.hasClass('grid-item')).toBeTruthy();
  });
});
