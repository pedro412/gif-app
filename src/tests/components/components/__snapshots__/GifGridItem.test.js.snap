// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <GifGridItem /> it should render correctly 1`] = `
<div
  className="grid-item"
>
  <img
    alt="Hola"
    src="https://algo.com"
  />
  <p>
    Hola
  </p>
</div>
`;
